core/data.go

```diff
@@ -554,61 +562,57 @@ func (pm *PaymentMetadata) UnmarshalDynamoDBAttributeValue(av types.AttributeVal
 		return fmt.Errorf("expected *types.AttributeValueMemberM, got %T", av)
 	}
 	pm.AccountID = m.Value["AccountID"].(*types.AttributeValueMemberS).Value
-	binIndex, err := strconv.ParseUint(m.Value["BinIndex"].(*types.AttributeValueMemberN).Value, 10, 32)
+	reservationPeriod, err := strconv.ParseUint(m.Value["ReservationPeriod"].(*types.AttributeValueMemberN).Value, 10, 32)
 	if err != nil {
-		return fmt.Errorf("failed to parse BinIndex: %w", err)
+		return fmt.Errorf("failed to parse ReservationPeriod: %w", err)
 	}
-	pm.BinIndex = uint32(binIndex)
+	pm.ReservationPeriod = uint32(reservationPeriod)
 	pm.CumulativePayment, _ = new(big.Int).SetString(m.Value["CumulativePayment"].(*types.AttributeValueMemberN).Value, 10)
 	return nil
```

Comment:
do we need to add salt?

Reply:
Do we need to store salt? I was thinking if salt is different then the blobHeaderHash would be different and we don't need to explicitly store it anywhere?

Reply:
Okay cool, I added Salt as part of this unmarshal function. If we decide to store salt, I can add it in a subsequent PR since it doesn't seem necessary for now

Reply:
sg lets add it in a follow up PR